kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -262,6 +405,7 @@ func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.Kaf
 			msg = fmt.Sprintf("Deployment created, error: %v", err)
 		}
 		r.Recorder.Eventf(src, corev1.EventTypeNormal, kafkaSourceDeploymentCreated, "%s", msg)
+		r.deployKedaScaledObject(ctx, ra, src)
```

Comment:
Isn't whether or not we're able to create a keda scaled object a separate concern to creating the receive adapter itself?

Perhaps it'd be better to call this function separately in the reconcile loop, after we get the resulting RA or error (one way or another), instead of trying to cover every single path in the `createReceiveAdapter` function

Reply:
I have moved the code and there is now only one call to do scaling after RA is created